doc/man3/STORE_open.pod

```diff
@@ -0,0 +1,121 @@
+read each available and supported object using STORE_load() as long as
+STORE_eof() hasn't been reached, and finish it off with STORE_close().
+
+The retrieved information is store in a B<STORE_INFO>, which is further
+described in L<STORE_INFO(3)>.
+
+=head2 Types
+
+B<STORE_CTX> is a context variable that holds all the internal
+information for STORE_open(), STORE_load(), STORE_eof() and
+STORE_close() to work together.
+
+=head2 Functions
+
+STORE_open() takes a uri or path B<uri>, password UI method
+B<ui_method> with associated data B<ui_data>, and post processing
+callback B<post_process> with associated data B<post_process_data>,
+opens a channel to the data located at that URI and returns a
+B<STORE_CTX> with all necessary internal information.
+The given B<ui_method> and B<ui_data_data> will be reused by all
+functions that use B<STORE_CTX> when interaction is needed.
+The given B<post_process> and B<post_process_data> will be reused by
+STORE_load() to manipulate or drop the value to be returned.
+
+STORE_ctrl() takes a B<STORE_CTX>, and command number B<cmd> and
+more arguments not specified here.  The available command numbers
+and arguments they each take depends on the loader that's used and
+is documented together with that loader.
+
+STORE_eof() takes a B<STORE_CTX> and checks if we've reached the end
+of data.
+
+STORE_load() takes a B<STORE_CTX>, tries to load the next available
+object and return it wrapped with  B<STORE_INFO>.
+
+STORE_close() takes a B<STORE_CTX>, closes the channel that was opened
+by STORE_open() and frees all other information that was stored in the
+B<STORE_CTX>, as well as the B<STORE_CTX> itself.
+
+=head1 SUPPORTED SCHEMES
+
+The basic supported scheme is B<file:>.
+Any other scheme can be added dynamically, using
+STORE_register_loader().
+
+=head1 NOTES
+
+When unsure whether a given string contains a simple file or directory
+reference, or if it's a full blown URI, the question is how to figure
+that out.  One way is to try STORE_open_file() and if that fails, try
+STORE_open().  The other way is the other way around.  Either way you
+choose, there are corner cases, F<file:/foo/bar/cookie.txt> might very
+will be a simple file reference on a system that supports the notion
+of volumes.
+
+This manual won't tell you which way is better, that's up to each
+application developer to decide on their own.  However, there are some
+tools that can be used together with STORE_open() to determine if any
+failure is caused by an unparsable URI, or if it's a different error
+(such as memory allocation failures); if the URI was unparsable, the
+top error will have the reason C<ERR_R_URI_LIB>, and if the URI was
+parsable but the scheme unregistered, the top error will have the
+reason C<STORE_R_UNREGISTERED_SCHEME>.  If you decide to use
+STORE_open() with STORE_open_file() as a fallback, those reasons can
+be good tools to decide if the fallback should be taken or not.
```

Comment:
Part of that is faulty, `ERR_R_URI_LIB` doesn't exist any more...  I'll have to think that one through.